### PR TITLE
[Feat] Account linking

### PR DESCRIPTION
**Pull request by igorosip0v**
This PR: 
- Adds forms for adding additional auth methods (Sign in with worldcoin / Email) directly from the logged in user UI. 

Logic: 
Once the user logs in with the one of auth methods, there will be an option on `/profile` page to add alternative method. Adding a method simply adds `email`/`world_id_nullifier` to the user's row in a Hasura user table.

On the next login (in `/login-callback)` we will check if the Hasura user has both `email` and `world_id_nullifier` we will try to fetch users from the `auth0` database using auth0 management API. And in case we have found two accounts (1: `email|<id>`, 2:`oauth2|worldcoin|<hash>`) we can link them using the `auth0Id` field from the Hasura `user` table to define the primary account. After linking accounts primary auth0 account will stay, and the second one will be removed from the users list and added as identity to the primary account.

NOTE: 
- This PR is not fully solves the issue, but it will be still an improvement, as I can see it. 
  - To make this work we will need to add some additional flow that will inform the user that it's worth connecting both auth methods, I think.
  - Also not fully clear what we should do in case if some of users actively using both auth method with different accounts right now.

- ‼️ Not fully tested
  - I didn't succeed in making sign in with worldcoin method working locally. 
    - It fails [here](https://github.com/worldcoin/developer-portal/blob/3f928f0513cd87a3357eb9aaa4b5de2a344d00ab/web/legacy/backend/kms.ts#L92-L97). I've decided not to spend much time on it and finish PR first.

---

<details>
<summary>Adding sign in with worldcoin</summary>

https://github.com/worldcoin/developer-portal/assets/89008845/83a47013-85c2-4d80-9bd5-3c76276a835e
</details>

<details>
<summary>Adding Email</summary>

https://github.com/worldcoin/developer-portal/assets/89008845/750f5b5d-ef83-4c56-a393-631d93f7885e

</details>

### CODE REVIEW

File: web/scenes/Portal/Profile/page/AuthMethodDialog/EmailForm/send-verification-email.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { ManagementClient, Passwordless } from "auth0";
+
+export const sendVerificationEmail = async (params: { email: string }) => {
+  if (
+    !process.env.AUTH0_CLIENT_ID ||
+    !process.env.AUTH0_CLIENT_SECRET ||
+    !process.env.AUTH0_DOMAIN
+  ) {
+    logger.error(
+      "Missing Auth0 environment variables. Impossible to send verification email.",
```
Comment (andy-t-wang) [inline web/scenes/Portal/Profile/page/AuthMethodDialog/EmailForm/send-verification-email.ts:13]:
```suggestion
      "Missing Auth0 environment variables.",
```

File: web/api/login-callback/index.ts [hunk 2]
```diff
@@ -149,6 +154,19 @@ export const loginCallback = withApiAuthRequired(async (req: NextRequest) => {
     );
   }
 
+  // ANCHOR: Auth0 account linking
+  if (user.world_id_nullifier && user.email && user.auth0Id) {
+    try {
+      await linkAccounts({
+        email: user.email,
```
[the remark this answers is not in the document]
Comment (igorosip0v) [reply web/api/login-callback/index.ts:161]: @andy-t-wang In theory it can be done right in the place where we are adding an additional auth method. But it will work only if the initial auth method was Sign in with worldcoin and we are adding an email method. In this case, we are [starting email verification](https://github.com/worldcoin/developer-portal/pull/666/files#diff-1c2dbfbdb511dcdb5296f07ad17bcaed3969764ea20d5693885021e28dbc055eR34) with auth0 and it will create an email auth0 account that can be linked to the primary auth0 account right after we did this. 

But in the case where email was primary auth method we can only simulate world id verification to get nullifier hash to put it in the hasura DB. We can't create worldcoin type auth0 account by API, it just throwing an error, when you try to do this AND also you can't just add worldcoin account identity directly to the primary email account. Because additional identity should be created from existing auth0 account. 

**TLDR;** So currently we are doing it on login because in case where email auth method is primary we need to create worldcoin type auth0 account first to connect it as identity to the primary account later. It can be done on login only, as I can see.

[the remark this answers is not in the document]
Comment (andy-t-wang) [reply web/api/login-callback/index.ts:161]: To be honest I think account linking only makes sense if another account already exists. This is like pre-emptive linking.

[the remark this answers is not in the document]
Comment (andy-t-wang) [reply web/api/login-callback/index.ts:161]: I guess in the ultimate ideal case now that I think about it we should have users link accounts that exist and merge the data. This feels hacky otherwise

**Comment by andy-t-wang** [comment]
Also we need to get designs for this

File: web/api/login-callback/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (igorosip0v) [reply web/api/login-callback/index.ts:161]: > makes sense if another account already exists
Ok, let's think that we will implement something that will merge accounts for the users that already have two accounts that were created before. 

What the flow should look like for the new user? Should new users create two accounts also and only then our app will suggest merging them?

[the remark this answers is not in the document]
Comment (andy-t-wang) [reply web/api/login-callback/index.ts:161]: Yeah I think that makes more sense. No pre-emptive merging. We let users merge two existing accounts. The challenge will be with permissions though  since this will require editing teams that you're not a part of

[the remark this answers is not in the document]
Comment (andy-t-wang) [reply web/api/login-callback/index.ts:161]: I think merging makes the most sense. For example Paolo has two accounts that need to be merged and are already created.

**Review by andy-t-wang — CHANGES_REQUESTED** [review]
Marking is as changes requested since we switch to merging and need designs just so we remember
